File: app/src/main/java/me/gilo/starter/di/AppModule.java
package me.gilo.starter.di;

import dagger.Module;
import dagger.Provides;
import me.gilo.starter.StarterApp;

import javax.inject.Singleton;

@Module
public class AppModule {

    StarterApp app;

    void AppModule(StarterApp application) {
        app = application;
    }

    @Provides
    @Singleton
    StarterApp providesApplication() {
        return app;
    }

}

File: app/src/main/java/me/gilo/starter/di/AppModule.kt
package me.gilo.starter.di

import dagger.Module
import dagger.Provides
import me.gilo.starter.StarterApp

import javax.inject.Singleton

@Module
class AppModule {

    internal lateinit var app: StarterApp

    internal fun AppModule(application: StarterApp) {
        app = application
    }

    @Provides
    @Singleton
    internal fun providesApplication(): StarterApp {
        return app
    }

}
